src/main/scala/gov/nih/nlm/ncbi/finagle/consul/catalog/ConsulCatalogResolver.scala

```diff
@@ -66,8 +74,13 @@ class ConsulCatalogResolver extends Resolver {
         val as = jsonToAddresses(parse(response.getContentString()))
         update() = Addr.Bound(as.map(Address(_)))
         val idx = response.headerMap.getOrElse("X-Consul-Index", "0")
+
+        Try(idx.toFloat).foreach(index => consulIndex = index)
+
         cycle(idx)
-      case t: Throw[_] => timer.doLater(Duration(1, TimeUnit.SECONDS)) {
+      case Throw(t) => timer.doLater(Duration(1, TimeUnit.SECONDS)) {
+        log.warning(t, s"Exception throw while querying Consul for service discovery")
+
         cycle("0")
```

Comment:
BTW reseting index to 0 on exception is almost never what we want
reeseting to 0 will return all servers in the next consul call , i'm not sure what happens in finagle then it gets a server as new in case it already has one registered ... does it ignore it ? does it remove/add it ?
anyway i can't imagine where keeping old index on exception can cause issues , on the contrary it should be much safer to keep the index. Index never goes lower in existing cluster it only increases.